internal/internal_task_handlers.go

```diff
@@ -230,12 +232,21 @@ func (eh *history) IsNextWorkflowTaskFailed() (isFailed bool, binaryChecksum str
 		nextEventType := nextEvent.GetEventType()
 		isFailed := nextEventType == enumspb.EVENT_TYPE_WORKFLOW_TASK_TIMED_OUT || nextEventType == enumspb.EVENT_TYPE_WORKFLOW_TASK_FAILED
 		var binaryChecksum string
+		var flags []sdkFlag
 		if nextEventType == enumspb.EVENT_TYPE_WORKFLOW_TASK_COMPLETED {
 			binaryChecksum = nextEvent.GetWorkflowTaskCompletedEventAttributes().BinaryChecksum
+			for _, flag := range nextEvent.GetWorkflowTaskCompletedEventAttributes().GetSdkMetadata().GetLangUsedFlags() {
+				f := sdkFlagFromUint(flag)
+				if !f.isValid() {
+					// If a flag is not recognized (value is too high or not defined), it must fail the workflow task
+					return false, "", nil, errors.New("could not recognize SDK flag")
```

Comment:
Hmm, I'm actually not doing this currently and probably should.

Reply:
oh that reminds me I wanted to add a test for this case, I'll do that quickly